test.js

```diff
@@ -235,3 +235,25 @@ test('erases wrapped lines', t => {
 
 	spinner.stop();
 });
+
+test('reset frameIndex when setting new spinner', async t => {
+	const stream = getPassThroughStream();
+	const output = getStream(stream);
+
+	const spinner = new Ora({
+		stream,
+		isEnabled: true,
+		spinner: {frames: ['foo', 'fooo']}
+	});
+
+	spinner.render();
+	t.is(spinner.frameIndex, 1);
+
+	spinner.spinner = {frames: ['baz']};
+	spinner.render();
+
+	stream.end();
+
+	t.is(spinner.frameIndex, 0);
+	t.regex(stripAnsi(await output), /foo baz/);
+});
```

Comment:
Can you add a test for it throwing: https://github.com/sindresorhus/ora/pull/95/files#diff-168726dbe96b3ce427e7fedce31bb0bcR57